src/psyclone/tests/psyir/frontend/fparser2_fortran_use_test.py

```diff
@@ -46,34 +46,67 @@
 from psyclone.psyir.frontend.fparser2 import Fparser2Reader
 from psyclone.psyir.nodes import KernelSchedule, Container
 from psyclone.psyir.symbols import ContainerSymbol, SymbolError, Symbol, \
-    DataSymbol, AutomaticInterface, INTEGER_SINGLE_TYPE
+    DataSymbol, AutomaticInterface, INTEGER_SINGLE_TYPE, ScalarType, \
+    RoutineSymbol
 
 
 def test_use_return(parser):
-    ''' Check the the Fparser correctly handles when a function uses a
-    return variable with a kind defined inside the function.'''
+    ''' Check the the Fparser frontend correctly handles when a function uses
+    a return variable with a kind defined inside the function.'''
     processor = Fparser2Reader()
     reader = FortranStringReader("real(rkind) function x()\n"
                                  "  use my_mod, only: rkind\n"
                                  "  x = 1.0_rkind\n"
                                  "end function x\n")
     ptree = parser(reader)
-    processor.generate_psyir(ptree)
+    psyir = processor.generate_psyir(ptree)
+    sym = psyir.children[0].symbol_table.lookup("x")
+    assert isinstance(sym, DataSymbol)
+    assert isinstance(sym.datatype, ScalarType)
+    assert sym.datatype.intrinsic == ScalarType.Intrinsic.REAL
```

Comment:
Please could you check the precision too? (`sym.datatype.precision.name == "rkind"`)

Reply:
Done